### ISSUE TITLE
Implementation discussion

### ISSUE THREAD

**Issue by drex44** [drex44/Streams#1]
Stack:
ReactJS/NextJS
NodeJS
MySQL/MongoDB

**Comment by KOLANICH** [comment]
Why not to just store the data on Wikipedia?

**Comment by BenjaminHoegh** [comment]
this may be useful: [rapidapi](https://rapidapi.com/utelly/api/utelly?utm_source=mashape&utm_medium=301)

**Comment by drex44** [comment]
@KOLANICH It looks like a good idea! But do they provide API to save data and access data?

**Comment by KOLANICH** [comment]
>But do they provide API to save data and access data?

They do provide API and a lib for using it (pywikipediabot it is called). The api allows searching all the pages containing specific templates or marked by specific cathegory.  To parse wiki-markup y9u will need `mwparserfromhell`. The main problem here is that the articles are freeform text. There are templates, but in that templates the values are mostly free-form text = non-machine-readable. This is a big problem if one wants to parse the data into a database. For example I have tried to create a script parsing the data from Wikipedia about chemical compounds (at first I found ChemSpider, but its terms are inacceptable, so I thought it may be a good idea to create an own free alternative without that damned restrictions) , and encountered the problem that I couldn't detect quantites units reliably even with the help of machine learning. The solution is obvious - go and manually fix all the wikipedia articles, but it is a manual operation and is likely needed by noone but me; taking in account count of articles about chemicals in WP, I have better ways to spend my time than fixing not-enough machine-readable info. 

For the purpose of this app manual operations are completely OK because it is meant to be used by multiple users, in the case of unparseable date the user is advised to fix it, and fixing it would update the article. Also there is no quantities involved except date. So IMHO using WP for this app should work.

So
1. an app periodically downloads a wp page about the series
2. the app scans for templates and tables.
3. the app finds the needed field in template or column in a table
4. the app preprocesses the value and tries to parse the date
5. the app compares the parsed date to cached one. if the date is greater than cached one, it means a new episode has been added, and then the app notifies its user.

**Comment by sbarfurth** [comment]
Localization is a major issue with Wikipedia. Not only are dates formatted differently, but sections will have different titles and location may differ. I feel like using Wikipedia as a TV series API is lackluster. Why not use something like [TMDb](https://www.themoviedb.org/documentation/api)?

**Adding to that:** It's free for development, since request counts are low. In Germany something like what we are talking about is available as [werstreamtes](https://www.werstreamt.es) (whostreamsit) and they use this API. They are a mix of curation and crowdsourcing.

**Comment by KOLANICH** [comment]
>Localization is a major issue with Wikipedia.

Not an issue at all, imho. For using WP as a DB or data storage for global apps English WP should be used. Localized names can be used via wikidata linking pages on different languages.

**Comment by drex44** [comment]
@sebastianbarfurth TMDB is a great resource to fetch. But to store new data, we cannot use TMDB.

**So how about we use WP and TMDB both?** 
We create a list of required fields. If these are not found on WP then we fetch it from TMDB(also store it on WP for future use). 
We also allow users to edit the content for correction or to add missing information.
++ @KOLANICH

**Comment by drex44** [comment]
@BenjaminHoegh It's good but paid.

**Comment by KobbyMmo** [comment]
Can we have a timeline?

**Comment by BenjaminHoegh** [comment]
well Netflix shouldn't be hard to make work, as they provide a [API](https://netflix.github.io/genie/docs/3.0.0/rest/)

**Comment by drex44** [comment]
@KobbyMmo yes! I will create it this weekend. Lately I have been busy so I couldn't create it.

**Comment by nishitmehta1** [comment]
Are there any updates regarding this?
